### PR TITLE
primaryKey usedRelationships

### PR DESCRIPTION
**Pull request by yelhouti**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by yelhouti** [comment]
@mshima I suspect that for this one you are the only one who could help :D

File: utils/entity.js [hunk 1]
```diff
@@ -266,8 +263,8 @@ function prepareEntityPrimaryKeyForTemplates(entityWithConfig, generator, enable
             get reference() {
               return fieldToReference(entityWithConfig, this);
             },
-            get otherEntityField() {
-              return field.otherEntityField || relationship.otherEntityField;
+            get usedRelationships() {
+              return [relationship, ...field.usedRelationships];
```
[the remark this answers is not in the document]
Comment (yelhouti) [reply utils/entity.js:267]: sure I ll do all of this

**Review by mshima — APPROVED** [review]
LGTM

**Comment by yelhouti** [comment]
@mshima thanks a lot, you rock

**Pull request merged**